paasta_tools/mesos_maintenance.py

```diff
@@ -614,7 +614,7 @@ def unreserve_all_resources(hostnames):
                 raise HTTPError("Failed unreserving all of the resources on %s (%s). Aborting." % (hostname, slave_id))
 
 
-def drain(hostnames, start, duration):
+def drain(hostnames, start, duration, reserve_resources=True):
     """Schedules a maintenance window for the specified hosts and marks them as draining.
     :param hostnames: a list of hostnames
     :param start: the time to start the maintenance, represented as number of nanoseconds since the epoch
```

Comment:
Doesn't an if statement need to be here on line 625?

Reply:
Oh duh 🤦‍♂️  thanks for spotting!